Kuzzle/Kuzzle.cs

```diff
@@ -228,39 +229,55 @@ public sealed class Kuzzle : IKuzzleApi, IKuzzle {
     internal void ResponsesListener(object sender, string payload) {
       Response response = Response.FromString(payload);
 
-      if (requests.ContainsKey(response.Room)) {
-        if (response.Error != null) {
-          if (response.Error.Message == "Token expired") {
-            EventHandler.DispatchTokenExpired();
-          }
+      if (!requests.ContainsKey(response.Room)) {
+        EventHandler.DispatchUnhandledResponse(response);
+        return;
+      }
 
-          requests[response.RequestId].SetException(
-            new Exceptions.ApiErrorException(response));
-        } else {
-          requests[response.RequestId].SetResult(response);
-        }
+      TaskCompletionSource<Response> task = requests[response.RequestId];
 
-        lock (requests) {
-          requests.Remove(response.RequestId);
+      if (response.Error != null) {
+        if (response.Error.Message == "Token expired") {
```

Comment:
Why don't use the error id instead of comparing the string?

Reply:
Doesn't work on all versions of kuzzle v1